### DISCUSSION TITLE
Condensation function: mapping and members attributes are hardly accessible.

### DISCUSSION THREAD

**Discussion by Moss4t** [networkx/networkx#5446] [Q&A] [answered]
In [this](https://networkx.org/documentation/stable/_modules/networkx/algorithms/components/strongly_connected.html#condensation) documentation, I found how to access mapping attribute on graph C: C.graph['mapping']. But still no idea how to access the dictionary for found SCCs and corresponding nodes (still researching). Please make the description more clear on how to use this function.
I'm a beginner in python, but I'm already thankful for the makers of the code. It helps a lot!

**Comment by MridulS** [comment, marked as answer]
Thank you for reporting this issue @Moss4t and yes we should do provide at least an example of how to extract information from the returned graph object.

A quick example for accessing the SCC and the corresponding nodes:
``` python
>>> G = nx.barbell_graph(4, 0)
>>> G.remove_edge(3, 4)
>>> G = nx.DiGraph(G)
>>> H = nx.condensation(G)
>>> H.nodes.data()
NodeDataView({0: {'members': {0, 1, 2, 3}}, 1: {'members': {4, 5, 6, 7}}})
>>> H.graph['mapping']
{0: 0, 1: 0, 2: 0, 3: 0, 4: 1, 5: 1, 6: 1, 7: 1}

```

**Reply by Moss4t** [reply]
I found these two solutions:

```
edges = []
edges.append((2,1))
edges.append((1,3))
edges.append((1,4))
edges.append((3,2))
edges.append((4,5))
Graph = networkx.DiGraph(edges)

dag = networkx.condensation(Graph)
mapping = dag.graph["mapping"]
print("mapping:", mapping)
output: mapping:  {5: 0, 4: 1, 1: 2, 2: 2, 3: 2}

members = networkx.get_node_attributes(dag, "members")
print("members:". members)
output: members:  {0: {5}, 1: {4}, 2: {1, 2, 3}}
```

I needed exactly these informations. It's just missing from the description.

**Reply by dschult** [reply]
This looks like a good example to add to the docstring for this function.
You can make the example more in the style we use for other docstrings by
1) using `nx.condensation` instead of spelling out networkx (we made it so `nx` is automagically imported when these examples are tested).
2) Call  the graph `G` (which I know is considered bad choice because it is a one letter variable -- but we do it throughout the docs)
2) put all the edges in a list to the `nx.DiGraph` constructor: `G = nx.DiGraph([(2, 1), (1, 3), (1, 4), (3, 2), (4, 5)])`
